department-of-veterans-affairs/caseflow
# Creation of test scenario form elements for UI

Resolves [Creation of remaining test scenario form elements for UI](https://jira.devops.va.gov/browse/APPEALS-55086)

# Description
Added ScenarioConfigurations component to include checkboxes for all the test scenarios.
If the scenario checkbox is selected, a dropdown for the Test Target Types and an input field for the Test Target ID will be displayed.
Added mappings for the test scenario and appropriate test target type in the LoadTestScenario.json file

## Acceptance Criteria
- [x] The UI for test configuration is updated
   - A new component named ScenarioConfiguration.jsx
      - The component is located in client/app/test/loadTest
      - The following elements are added:
         - Scenario Groups:
            - A list of checkboxes of k6 scenario groups that are to be executed in a test.
            - A searchable dropdown associated with the test scenario for target type
            - A related input field to capture the id of the target
               - The inputted id will be the id that is inputted in the url of the get/post request
      - The new component is imported into LoadTest.jsx
    - A constants file is added 
       - The file is named LoadTestScenarios.js
       - The file is located in client/app/constants
       - The file contains
          - the test names of all scenarios being run
          - an array of target types (nullable)

## Testing Plan
1. Go to [http://localhost:3000/test/load_tests](http://localhost:3000/test/load_tests)
2. Select the checkboxes and verify that the Target Type dropdown and Target Type ID fields are shown for each checkbox
3. Verify the values in the Target Type dropdwon correspond to the expected values in the LoadTestScenario.json file
4. For the "schedule", "search", and "queue" checkboxes, no Target Type dropdown or Target Type ID fields should be displayed when they are checked


# Frontend
## User Facing Changes
 - [x] [Wireframes for the change](https://www.figma.com/proto/xBpCzh8wne71uUCLmZGlmi/Configuration-Of-Test-Targets?page-id=40%3A854&node-id=46-4122&viewport=351%2C-45%2C0.27&t=8BQ4ieD4V6SMtAVT-1&scaling=scale-down&content-scaling=fixed&starting-point-node-id=46%3A4122)
